java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java

```diff
@@ -160,25 +164,30 @@ protected void selectVerticalAutoTickUnit(Graphics2D g2, Rectangle2D dataArea, R
         TickUnit unit1 = tickUnits.getCeilingTickUnit(this.getTickUnit());
 
         //Smallest tick distance could be at either end
-        final double unitHeightUpper = Math.abs(this.lengthToJava2D(getRange().getUpperBound(), dataArea, edge) - this.lengthToJava2D(getRange().getUpperBound() - unit1.getSize(), dataArea, edge));
-        final double unitHeightLower = Math.abs(this.lengthToJava2D(getRange().getLowerBound(), dataArea, edge) - this.lengthToJava2D(getRange().getLowerBound() + unit1.getSize(), dataArea, edge));
+        final double unitHeightUpper = Math.abs(this.lengthToJava2D(Math.abs(getRange().getUpperBound()), dataArea, edge) - this.lengthToJava2D(Math.abs(getRange().getUpperBound() - unit1.getSize()),
+                dataArea,
+                edge));
+        final double unitHeightLower = Math.abs(this.lengthToJava2D(Math.abs(getRange().getLowerBound()), dataArea, edge) - this.lengthToJava2D(Math.abs(getRange().getLowerBound() + unit1.getSize()),
+                dataArea, edge));
         double unitHeight = Math.min(unitHeightLower, unitHeightUpper);
 
         double guess = unit1.getSize();
-        if(unitHeight > 0.0D) {
+        if (unitHeight > 0.0D) {
             guess = tickLabelHeight / unitHeight * unit1.getSize();
         }
 
-        NumberTickUnit unit2 = (NumberTickUnit)tickUnits.getCeilingTickUnit(guess);
+        NumberTickUnit unit2 = (NumberTickUnit) tickUnits.getCeilingTickUnit(guess);
 
-        final double unit2HeightUpper = Math.abs(this.lengthToJava2D(getRange().getUpperBound(), dataArea, edge) - this.lengthToJava2D(getRange().getUpperBound() - unit2.getSize(), dataArea, edge));
-        final double unit2HeightLower = Math.abs(this.lengthToJava2D(getRange().getLowerBound(), dataArea, edge) - this.lengthToJava2D(getRange().getLowerBound() + unit2.getSize(), dataArea, edge));
+        final double unit2HeightUpper = Math.abs(this.lengthToJava2D(Math.abs(getRange().getUpperBound()), dataArea, edge) - this.lengthToJava2D(Math.abs(getRange().getUpperBound() - unit2.getSize()),
+                dataArea, edge));
+        final double unit2HeightLower = Math.abs(this.lengthToJava2D(Math.abs(getRange().getLowerBound()), dataArea, edge) - this.lengthToJava2D(Math.abs(getRange().getLowerBound() + unit2.getSize()),
```

Comment:
Is it expensive to call getRange() multiple times?  Should it just be called once and used multiple times?

Reply:
Is it expensive to call getRange() multiple times?  Should it just be called once and used multiple times?

Also, the abs calls around the range bounds seem supect.

Reply:
Cached getRange().

I put Math.abs to avoid negative bounds affecting the length computation while transformations applied as it came across while working on this fix. Now that the range computation makes sure that invisible points won't be visible, I can take those off and check again.